### PR TITLE
Add in CRT-Hit match filling to CAF. Refactor time reference of CRT. 

### PR DESCRIPTION
**Pull request by gputnam**
This makes the CRT time referencing more similar to how it works in CAFMaker for the PMTs and uses the newer trigger information.

This PR is coupled to changes in https://github.com/SBNSoftware/sbncode/pull/314, so I made this PR on top of that branch. 

I would recommend merging that PR first (at which point the first commit of this PR will be merged) and then proceeding with this branch.

### CODE REVIEW

**Review by PetrilloAtWork — CHANGES_REQUESTED** [review]
Beside the usual comments: I don't fully get the business of the CRT reference in MC.
It is foreseeable that MC will get its absolute trigger timestamp in the near future (based on the event time, based on the production time). That may change things here too.
Mind to comment on the topic, and explain the choice here?
I expect this to turn into a "let's open a ticket to remind us the problem, and move on with this PR", but I would like to make sure that's the case.

File: sbncode/CAFMaker/CAFMaker_module.cc [hunk 1]
```diff
@@ -417,6 +419,46 @@ void CAFMaker::BlindEnergyParameters(StandardRecord* brec) {
...
+    for (SRTrack &t: s.reco.trk) {
+      t.crthit.hit.time += CRT_reference_time;
+    }
+  }
+  for (SRTrack &t: rec.reco.trk) {
+    t.crthit.hit.time += CRT_reference_time;
+  }
+
+  // TODO: fix more?
+  // Tracks?
```
[the remark this answers is not in the document]
Comment (gputnam) [reply sbncode/CAFMaker/CAFMaker_module.cc:458]: This comment is just there as a note to future developers.

File: sbncode/CAFMaker/CAFMaker_module.cc [hunk 2]
```diff
@@ -246,6 +245,9 @@ class CAFMaker : public art::EDProducer {
 
   void InitVolumes(); ///< Initialize volumes from Gemotry service
 
+  void ReferencePMTTimes(StandardRecord &rec, double PMT_refernce_time);
+  void ReferenceCRTTimes(StandardRecord &rec, double CRT_reference_time);
```
[the remark this answers is not in the document]
Comment (gputnam) [reply sbncode/CAFMaker/CAFMaker_module.cc]: Fixed.

**Comment by gputnam** [comment]
I've added the new commit from https://github.com/SBNSoftware/sbncode/pull/314 to this PR and addressed comments.

File: sbncode/CAFMaker/FillTrigger.cxx [hunk 3]
```diff
@@ -4,23 +4,29 @@
...
   {
     uint64_t beam_ts = 0;
     uint64_t trigger_ts = 0;
-    triggerInfo.emplace_back();
-    triggerInfo.back().global_trigger_time = addltrig_info.triggerTimestamp;
-    triggerInfo.back().beam_gate_time_abs = addltrig_info.beamGateTimestamp;
+    triggerInfo.global_trigger_time = addltrig_info.triggerTimestamp;
+    triggerInfo.beam_gate_time_abs = addltrig_info.beamGateTimestamp;
     beam_ts = addltrig_info.beamGateTimestamp;
     trigger_ts = addltrig_info.triggerTimestamp;
```
Comment (PetrilloAtWork) [inline sbncode/CAFMaker/FillTrigger.cxx:15]: I am surprised that the compiler does not complain about these two variables being "set but not used" or something like that. I suggest they are just removed (including their definitions above at lines 10-11).
```suggestion
```

**Review by PetrilloAtWork — APPROVED** [review]
I still don't get the business of the absolute reference in MC. I admittedly don't know about the CRT system, so I trust the due thinking has been thought on that.
For the rest, one C++ comment left, but good to go.

File: sbncode/CAFMaker/FillTrigger.cxx (hunk 3 above)
Comment (gputnam) [reply sbncode/CAFMaker/FillTrigger.cxx]: Fixed.

**Comment by PetrilloAtWork** [comment]
@gputnam looking at Clang compilation failures:
* `candidate function template not viable: no known conversion from 'const art::InputTag' to 'const std::string'`: apparently Clang does not want to implicitly convert the tag into a string; you can do that explicitly (why is GCC happy with that?!): explicit conversion should fix that: `GetByLabelStrict(evt, fParams.TriggerLabel().encode(), extratrig_handle);` (`GetByLabelStrict()` should take a `art::InputTag`, but that's for another PR).
* `no known conversion from 'caf::SRTrigger' to 'const std::__1::vector<caf::SRTrigger, std::__1::allocator<caf::SRTrigger> >' for 1st argument` here I am not so sure... you changed somewhere a `vector<caf::SRTrigger>` into a single `caf::SRTrigger`, supposedly you did consistently and this is just a merge/branch problem.

**Comment by gputnam** [comment]
I've fixed the build errors -- can we try again?

**Pull request merged**